sql/core/src/main/scala/org/apache/spark/sql/internal/SharedState.scala

```diff
@@ -69,6 +71,29 @@ private[sql] class SharedState(val sparkContext: SparkContext) extends Logging {
   val jarClassLoader = new NonClosableMutableURLClassLoader(
     org.apache.spark.util.Utils.getContextOrSparkClassLoader)
 
+  /**
+   * Add a global-scoped jar
+   */
+  def addJar(path: String): Unit = {
+    if (sparkContext.conf.get(CATALOG_IMPLEMENTATION) == "hive") {
+      // Hive metastore supports custom serde.
+      externalCatalog.addJar(path)
```

Comment:
Also used by run-time function loading when Hive support is enabled. 

If Hive support is not enabled, `ADD JAR` is the only caller of `SharedState.addJar`. It is just a SQL interface of `sparkContext.addJar`. I am not sure whether end users are using `ADD JAR` for `sparkContext.addJar`. Thus, I am hesitant to make any external behavior change in this PR.